nmrglue/fileio/varian.py

```diff
@@ -1895,25 +1895,26 @@ def get_parameter(f):
     line = f.readline().decode().split()
 
     dic["name"] = line[0]
-    dic["subtype"] = line[1]
-    dic["basictype"] = line[2]
-    dic["maxvalue"] = line[3]
-    dic["minvalue"] = line[4]
-    dic["stepsize"] = line[5]
-    dic["Ggroup"] = line[6]
-    dic["Dgroup"] = line[7]
-    dic["protection"] = line[8]
-    dic["active"] = line[9]
-    dic["intptr"] = line[10]
+    dic["subtype"] = int(line[1])
+    dic["basictype"] = int(line[2])
+    dic["maxvalue"] = float(line[3])
+    dic["minvalue"] = float(line[4])
+    dic["stepsize"] = float(line[5])
+    dic["Ggroup"] = int(line[6])
+    dic["Dgroup"] = int(line[7])
+    dic["protection"] = np.uint32(line[8])
+    dic["active"] = int(line[9])
+    dic["intptr"] = int(line[10])
 
     # read in the values of the parameter
     line = f.readline().decode()
     num = int(line.split()[0])
     values = []
 
-    if dic["basictype"] == "1":     # real values, only one line
-        values = line.split()[1:]
-    elif dic["basictype"] == "2":   # strings, may have multiple lines
+    if dic["basictype"] == 1:     # real values, only one line
+        values = list(map(np.float, line.split()[1:]))
+        values = [int(v) if v.is_integer() else v for v in values]
```

Comment:
I do not like the use of np.float here.  I would prefer these values be stored as Python floats without a cast to integers.  Nothing in the procpar file specifies these as ints.